android/fakedapp/src/main/java/com/solana/mobilewalletadapter/fakedapp/MainViewModel.kt

```diff
@@ -148,268 +135,235 @@ class MainViewModel(application: Application) : AndroidViewModel(application) {
                 )
             }
             authorized = true
-        } catch (e: IOException) {
-            Log.e(TAG, "IO error while sending authorize", e)
-        } catch (e: JsonRpc20Client.JsonRpc20RemoteException) {
-            when (e.code) {
-                ProtocolContract.ERROR_AUTHORIZATION_FAILED -> Log.e(TAG, "Not authorized", e)
-                else -> Log.e(TAG, "Remote exception for authorize", e)
+        } catch (e: ExecutionException) {
```

Comment:
I still think this ViewModel class has a lot of exception spam. I'd like to see, um, one method where all the common exceptions are enumerated, and then if each call has its own exceptions, they could be put here.

But also, documenting exceptions by way of tests would be an option too. Kotlin's unchecked exceptions mean we don't have to do this.

Reply:
The full exception catching blocks were here for development during debugging. I'll centralize the common exceptions in a future PR. I wanted to keep the same structure with this PR, to ease the cognitive review burden and keep it focused on the API changes to `Scenario`